lab-lee/test/duck-route-test.js

```diff
@@ -0,0 +1,260 @@
+'use strict';
+
+const PORT = process.env.PORT|| 3000;
+process.env.MONGODB_URI = 'mongodb://localhost/ducktest';
+
+const expect = require('chai').expect;
+const request = require('superagent');
+const Fowl = require('../model/fowl.js');
+const Duck = require('../model/duck.js');
+
+require('../server.js');
+
+const url = `http://localhost:${PORT}`;
+
+const exampleDuck = {
+  name: 'Jeff',
+  color: 'blue',
+  feathers: '15',
+};
+
+const exampleFowl = {
+  name: 'Fowls1',
+  timestamp: new Date(),
+};
+
+
+describe('testing duck routes', function() {
+
+  describe('testing POST requests', function() {
+
+    describe('with valid fowl id and duckBody', function() {
+
+      before( done => {
+        new Fowl(exampleFowl).save()
+        .then( fowl => {
+          this.tempFowl = fowl;
+          done();
+        })
+        .catch(done);
+      });
+
+      after( done => {
+        Promise.all([
+          Fowl.remove({}),
+          Duck.remove({}),
+        ])
+        .then(() => done())
+        .catch(done);
+      });
+
+      it('should return a duck', done => {
+        request.post(`${url}/api/fowl/${this.tempFowl._id}/duck`)
+        .send(exampleDuck)
+        .end((err, res) => {
+          if(err) return done(err);
+          expect(res.status).to.equal(200);
+          expect(res.body.name).to.equal(exampleDuck.name);
+          expect(res.body.fowlID).to.equal(this.tempFowl._id.toString());
+          done();
+        });
+      });
+
+      describe('testing POST requests with invalid body or no body provided', function() {
+
+        before( done => {
```

Comment:
You can set this all up in the outer POST test describe block and then you don't have to repeat yourself (as long as you maintain lexical scope)!